### DISCUSSION TITLE
Download page and issue 772

### DISCUSSION THREAD

**Discussion by terminator356** [muse-sequencer/muse#1159] [Team Posts]
What is the purpose of this extra file on the download page?
I cannot provide support for it because it does not reflect the **actual source tree**.
It will cause confusion, as has just been demonstrated in issue 772.
His logs clearly indicate that it was possible he could have been using
 that downloadable file. Which caused a lot of confusion.

The reporter now says he is using the git master.
Which means that after all this, he's not building properly because
 there's no muse3 folder containing his build folder.

Now I have to go and correct him again.

**Comment by spamatica** [comment]
I'm a bit confused, seems you are implying there is /one/ extra file?
The muse-3.1.tar.gz is the release, same structure as it has had for a long time.
The Source code* files however seems to be something github always puts there. Not sure if they can be hidden. I think they are supposed to contain a clean copy of the branch that the tag was created in.

And yes, as they contain a folder that isn't in the ordinary tarball it does cause a bit of confusion. 
Though people (and us devs) build from git all the time so I didn't think it was that big of a deal.

Having the build dir inside the muse3 folder is actually not a requirement at all, the build folder can be placed pretty much anywhere. I often have the build folder on the same level as the git repo and create the build files with "cmake -DCMAKE_BUILD_TYPE=debug ../muse/muse3" from the build folder

I haven't looked that closely at the logs but my bet is on that this is some tool chain complication on arm-linux.

**Comment by terminator356** [comment]
There are three files, two of which accurately reflect the source tree,
 the other muse-3.1.tar.gz does not.
I **did not realize until now** that all our releases on SF have had the muse3 folder missing.
Now that we have moved the releases to github, having these three files listed
 is not good, it's going to cause headaches.
This is not a good situation. Now I have to ask everyone with build troubles which
 actual file they downloaded? As I mistakenly tell them that there should be a muse3 folder?

We'll roll with this for now, but with future releases we'll have to change something here.

All of this may have made sense when the releases were on SF, but not now.
Honestly I feel that only the two Source files should be there.
The lack of muse3 folder might make our build instructions inaccurate,
 or my verbal instructions inaccurate.

(The muse3 folder was put there to keep the github front page from being cluttered with
 dozens of folders and files. Don't remove it without adequate warning, because it will seriously
 mess up any work-in-progress branches from being able to update cleanly.)

I just looked at the Jack github repo download page and they also have three files.
But all three are identical. Why bother having the first file at all?

BTW There is an extra file in the release: CMakeLists.txt.user, which is a QtCreator file.
Was that intentional?

**Comment by terminator356** [comment]
Today was a frustrating day, things were going wrong.
I overreacted.

You know, I have a couple of unpacked release trees sitting in my source folder.
I have in fact seen the release structure before.
For some crazy reason I never put two and two together to realize that's how the releases are.
I'm not sure what the heck I thought. Maybe that they were was just old or something?

Now that the releases are in github I kind of went all alarmist on y'all.
I thought the Source files were the real releases.
I thought the release file was something you were working on for AppImage or FlatPak,
 'cause I thought I read you talking about it.

Comedy of errors.

**Comment by spamatica** [comment]
Den ons 26 feb. 2020 kl 04:52 skrev Tim <notifications@github.com>:

> Today was a frustrating day, things were going wrong.
> I overreacted.
>
> You know, I have a couple of unpacked release trees sitting in my source
> folder.
> I have in fact seen the release structure before.
> For some crazy reason I never put two and two together to realize that's
> how the releases are.
> I'm not sure what the heck I thought. Maybe that they were was just old or
> something?
>
> Now that the releases are in github I kind of went all alarmist on y'all.
> I thought the Source files were the real releases.
> I thought the release file was something you were working on for AppImage
> or FlatPak,
> 'cause I thought I read you talking about it.
>
> Comedy of errors.
>
No worries. It is and was a bit confusing.
It's a valid point about the jack releases, I guess the only real reason is
to get a file that has a name that adheres to how source releases are
generally named.

We could go back to the sourceforge page but in the long run I think we are
probably better off on github.
Also, github allows direkt links to the packages so we could (and maybe
should) have a direct link when publicizing releases instead of pointing to
the release page, then it would be clear which file is the right one.
